### PR TITLE
Introduce new status for automatic translations

### PR DESCRIPTION
**Pull request by SteffiStuffel**
### Short description
Added new machine_translated status to translation status.
Added bot icon to show if a translation is machine_translated
Works for SUMM.AI, XLIFF-Imports can be marked with a checkbox as machine translated, DeepL is also implemented, but I couldn't test it, as I don't know how to activate it.
Works for pages, events and pois

### Proposed changes
Removed unnecessary verbose name in AbtractContentTranslations
Moved shared Meta fields of translation forms to customContentModelFrom

### Resolved issues
Fixes: #1883 

__________________________________________________

[Pull Request Review Guidelines](https://digitalfabrik.github.io/integreat-cms/pull-request-review-guide.html)

### CODE REVIEW

File: integreat_cms/locale/de/LC_MESSAGES/django.po [hunk 1]
```diff
@@ -2084,39 +2087,20 @@ msgstr ""
...
-msgstr "wird derzeit übersetzt"
+msgid ""
+"Flag to indicate a translation is being updated by an external translator"
+msgstr "Feld, das anzeigt, dass eine Übersetzung extern aktualisiert wird"
 
 #: cms/models/abstract_content_translation.py
 msgid ""
-"Flag to indicate a translation is being updated by an external translator"
+"Flag to indicate,if a translations is machine translated and needs review"
 msgstr "Feld, das anzeigt, dass eine Übersetzung extern aktualisiert wird"
```
Comment (timoludwig) [inline integreat_cms/locale/de/LC_MESSAGES/django.po:2098]:
```suggestion
msgstr "Feld, das anzeigt, dass eine Übersetzung maschinell übersetzt wurde"
```

File: integreat_cms/cms/models/abstract_content_translation.py [hunk 2]
```diff
@@ -33,25 +33,27 @@ class AbstractContentTranslation(AbstractBaseModel):
...
     )
     currently_in_translation = models.BooleanField(
         default=False,
-        verbose_name=_("currently in translation"),
         help_text=_(
             "Flag to indicate a translation is being updated by an external translator"
         ),
     )
+    machine_translated = models.BooleanField(
+        default=False,
```
Comment (timoludwig) [inline integreat_cms/cms/models/abstract_content_translation.py:49]:
```suggestion
        default=False,
        verbose_name=_("machine translated"),
```

File: integreat_cms/cms/migrations/0054_removed_unnecessary_verbose_names_from_pagetranslation.py [hunk 3]
```diff
@@ -0,0 +1,258 @@
+# Generated by Django 3.2.16 on 2022-11-23 11:08
```
Comment (timoludwig) [inline integreat_cms/cms/migrations/0054_removed_unnecessary_verbose_names_from_pagetranslation.py:1]: I'd suggest to remove this migration again.

**Review by timoludwig — COMMENTED** [review]
Thanks a lot, really cool feature! :+1:

Just a few minor comments, in general it works exactly as expected :muscle:

File: integreat_cms/cms/constants/translation_status.py [hunk 4]
```diff
@@ -3,16 +3,12 @@
...
 
-#: Up to date
 UP_TO_DATE = "UP_TO_DATE"
-#: In translation
 IN_TRANSLATION = "IN_TRANSLATION"
-#: Outdated
 OUTDATED = "OUTDATED"
-#: Fallback
 FALLBACK = "FALLBACK"
-#: Missing
```
[the remark this answers is not in the document]
Comment (SteffiStuffel) [reply integreat_cms/cms/constants/translation_status.py:14]: I also had a look a the docs and as these constants are also part of the "Choices" and are documented there, I thought it is fine to remove these there. Reverted anyways.

File: integreat_cms/cms/models/abstract_content_translation.py [hunk 5]
```diff
@@ -17,7 +17,7 @@ class AbstractContentTranslation(AbstractBaseModel):
     Data model representing a translation of some kind of content (e.g. pages or events)
     """
 
-    title = models.CharField(max_length=1024, verbose_name=_("title"))
+    title = models.CharField(max_length=1024)
```
[the remark this answers is not in the document]
Comment (SteffiStuffel) [reply integreat_cms/cms/models/abstract_content_translation.py:20]: I removed them, where they are the same as the django defaults, but you are right, removing them breaks the translation.

**Review by timoludwig — APPROVED** [review]
Awesome, works like a charm :pray:

**Comment by timoludwig** [comment]
> One more thing I noticed. Since I updated the dependencies yesterday I run into an issue when I click on pages, pois or events. Maybe this is again just on me or maybe rebasing and updating the dependencies might already solve this issue :)

@JoeyStk This has nothing to do with dependencies, but with the state of your local database. You changed from another branch to this branch without applying the migrations which are needed for this branch. So the error is completely expected. For the future, just keep in mind those ground rules:
- When changing to another branch where the Pipfile.lock or package-lock.json differs, you need to run `./dev-tools/install.sh` again
- When changing from the default branch to a feature branch with migrations, either restart the server or run `./dev-tools/migrate.sh` in another terminal
- When changing from a non-default branch with migrations (which were applied) to another branch with different (conflicting) migrations, please reset your database with the `./dev-tools/prune_database.sh` script and restart the server

**Comment by JoeyStk** [comment]
Thank you!

Now I was able to test it also for DeepL and it works as expected :muscle: !

**Pull request merged**